rest-api/dao/questionnaire_response_dao.py

```diff
@@ -200,8 +201,11 @@ def _update_participant_summary(
         if code:
           summary_field = QUESTION_CODE_TO_FIELD.get(code.value)
           if summary_field:
-            something_changed = self._update_field(participant_summary, summary_field[0],
-                                                   summary_field[1], answer)
+            if something_changed:
+              self._update_field(participant_summary, summary_field[0], summary_field[1], answer)
+            else:
```

Comment:
Make sure once the `something_changed` was set to TRUE, it will not overwrite back the FALSE.